fixtures/coverall/tests/bindings/test_coverall.py

```diff
@@ -441,9 +441,28 @@ def test_path(self):
         py_node.set_parent(None)
         traits[0].set_parent(None)
 
+    def test_struct_traits(self):
+        # A struct, but also a trait.
+        node = Node("node")
+        # parent is a "raw" Rust object
+        self.assertEqual(node.describe(), "Node: name=Some(\"node\"), parent=true")
+        # Unfortunately, this trait object gets re-wrapped as it's passed back to Rust, because
+        # we can't easily know it's alreqdy the correct type.
+        # Nasty to rely on the repr of the inner mutex, but it neatly demonstrates the issue.
```

Comment:
- Maybe also point people towards the `proc-macro` fixture so that they can see the normal usage without all the gotchas that you need to mention here.
- This re-wrapping also happens for regular trait objects right?  Do you think the handle changes I've been hoping to introduce do could address both cases?

Reply:
> This re-wrapping also happens for regular trait objects right? Do you think the handle changes I've been hoping to introduce do could address both cases?

It does, yes. It will be mildly more complicated to fix here, only because we have a handle/pointer to a struct that we need to convert to a handle/pointer to a dyn trait. I think we'd also need an FFI function something like `fn to_trait(ob: Arc<StructName>) -> Arc<dyn TraitName>`, which is just a trivial `Arc::new(*ob as dyn TraitName)` or similar, then some other magic in the bindings that knows to use this at the right time.

Certainly fixing the existing times we do this wrapping would be a pre-req though!